pulpcore.fc

```diff
@@ -1,16 +1,21 @@
-/usr/lib/pulp/bin/gunicorn		--	gen_context(system_u:object_r:pulpcore_exec_t,s0)
+/etc/pulp(/.*)?			gen_context(system_u:object_r:pulpcore_etc_t,s0)
```

Comment:
Not everything in `/etc/pulp` belongs to Pulp 3. On migrations there's also Pulp 2 content there.

Reply:
If we knew what kind of files can be there, we could assign different types. With this line, all files have the same default private type `pulpcore_etc_t`. Is this a problem?

Reply:
I can't find the git repo for the old pulp-selinux package, but what I see today:
```console
# ls -lZ /etc/pulp
drwxr-xr-x. root   root   system_u:object_r:httpd_sys_content_t:s0 content
-rw-r--r--. root   root   system_u:object_r:httpd_sys_content_t:s0 repo_auth.conf
drwxr-xr-x. root   root   system_u:object_r:httpd_sys_content_t:s0 server
-rw-------. apache apache system_u:object_r:httpd_sys_content_t:s0 server.conf
-rw-r-----. root   pulp   system_u:object_r:httpd_sys_content_t:s0 settings.py
-rw-r--r--. root   root   system_u:object_r:httpd_sys_content_t:s0 streamer.conf
drwxr-xr-x. root   root   system_u:object_r:httpd_sys_content_t:s0 vhosts80
```

This comes from:
```console
# semanage fcontext --list | grep pulp
/etc/pulp(/.*)?                                    all files          system_u:object_r:httpd_sys_content_t:s0 
/var/run/pulp(/.*)?                                all files          system_u:object_r:pulp_var_run_t:s0 
/etc/pki/pulp(/.*)?                                all files          system_u:object_r:pulp_cert_t:s0 
/var/lib/pulp(/.*)?                                all files          system_u:object_r:httpd_sys_rw_content_t:s0 
/var/log/pulp(/.*)?                                all files          system_u:object_r:httpd_sys_rw_content_t:s0 
/var/cache/pulp(/.*)?                              all files          system_u:object_r:pulp_var_cache_t:s0 
/var/lib/pulp/tmp(/.*)?                            all files          system_u:object_r:pulpcore_var_lib_t:s0 
/var/run/pulpcore*(/.*)?                           all files          system_u:object_r:pulpcore_var_run_t:s0 
/usr/share/pulp/wsgi(/.*)?                         all files          system_u:object_r:httpd_sys_content_t:s0 
/var/lib/pulp/assets(/.*)?                         all files          system_u:object_r:pulpcore_var_lib_t:s0 
/var/lib/pulp/upload(/.*)?                         all files          system_u:object_r:pulpcore_var_lib_t:s0 
/var/run/pulpcore-api(/.*)?                        all files          system_u:object_r:pulpcore_var_run_t:s0 
/var/lib/pulp/artifact(/.*)?                       all files          system_u:object_r:pulpcore_var_lib_t:s0 
/var/run/pulpcore-content(/.*)?                    all files          system_u:object_r:pulpcore_var_run_t:s0 
/var/run/pulpcore-resource-worker[0-9]*(/.*)?      all files          system_u:object_r:pulpcore_var_run_t:s0 
/var/run/pulpcore-resource-manager(/.*)?           all files          system_u:object_r:pulpcore_var_run_t:s0 
/usr/lib/pulp/bin/rq                               regular file       system_u:object_r:pulpcore_exec_t:s0 
/usr/bin/pulp_streamer                             regular file       system_u:object_r:streamer_exec_t:s0 
/usr/lib/pulp/bin/gunicorn                         regular file       system_u:object_r:pulpcore_exec_t:s0 
/usr/local/lib/pulp/bin/rq                         regular file       system_u:object_r:pulpcore_exec_t:s0 
/usr/local/lib/pulp/bin/gunicorn                   regular file       system_u:object_r:pulpcore_exec_t:s0 
```

Note the first line - this is an exact duplicate of that. In Satellite the migration plan is to install both in parallel and online migrate from 2 to 3. So both must be able to operate in parallel.

You're more familiar with SELinux so I hope you can tell me what SELinux does if there are 2 patterns that are exactly the same, but I can't imagine that works well.

Reply:
For each path there can be only one rule, just more specific rules have priority over regexps.
I can drop the pulpcore_etc_t support completely, but on the system I have access to there is no default rule so the files will get etc_t.

Reply:
IMHO the cleanest solution is to use `/etc/pulpcore` and place `settings.py` there, but that's something that needs more planning. Should we make it more specific and explicitly list `/etc/pulpcore/settings.py`? AFAIK that's the only file in it actually needs. That probably warrants a comment, something like `# More specific to avoid conflicts with Pulp 2's SELinux policy`.

Reply:
From SELinux POV no need for a different directory, explicit path would work. There are also 2 keys there.